guides/source/sign_up_and_settings.md

```diff
@@ -0,0 +1,1658 @@
+
+We can add `attr_readonly` in our model to protect the attribute from updates.
+
+```ruby#5
+class User < ApplicationRecord
+  has_secure_password
+  has_many :sessions, dependent: :destroy
+
+  attr_readonly :admin
+
+  # ...
+```
+
+When admin is readonly, we have to directly update this in the database instead of using Active Record.
+
+Rails has a command called `dbconsole` that will open a database console where we can directly interact with the database using SQL.
+
+```bash
+$ rails dbconsole
+SQLite version 3.43.2 2023-10-10 13:08:14
+Enter ".help" for usage hints.
+sqlite>
+```
+
+In the SQLite prompt, we can update the admin column for a record using an `UPDATE` statement and using `WHERE` to filter to a single user ID.
+
+```sql
+UPDATE users SET admin=true WHERE users.id=1;
+```
+
+To close the SQLite prompt, enter the following command:
+
+```
+.quit
+```
+
+Viewing All Users
+-----------------
+
+As a store admin, we will want to view and manage users for customer support, marketing, etc.
+
+First, we'll need to add a route in the settings namespace for users in `config/routes.rb`.
+
+```ruby#2
+namespace :settings do
+  resource :account, only: [ :show, :destroy ]
+  resource :email, only: [ :show, :update ]
+  resource :password, only: [ :show, :update ]
+  resource :profile, only: [ :show, :update ]
+
+  # Admins only
+  resources :users
+
+  root to: redirect("/settings/profile")
+end
+```
+
+### Adding Admin Only Access
+
+The controller for users should be accessible to admins only. Before we create that controller, let's add a class method to the `Authentication` module to restrict access to admins only.
+
+Open `app/controllers/concerns/authentication.rb` and add:
+
+```ruby#19-21
+module Authentication
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :require_authentication
+    helper_method :authenticated?
+  end
+
+  class_methods do
+    def allow_unauthenticated_access(**options)
+      skip_before_action :require_authentication, **options
+    end
+
+    def unauthenticated_access_only(**options)
+      allow_unauthenticated_access **options
+      before_action -> { redirect_to root_path if authenticated? }, **options
+    end
+
+    def admin_access_only(**options)
+      before_action -> { redirect_to root_path, alert: "You aren't allowed to do that." unless authenticated? && Current.user.admin? }, **options
+    end
+
+    # ...
+```
+
+Adding this to the Authentication module allows us to use this method in any controller to restrict actions to admin only access.
+
+### Users Controller & Views
+
+We can then use this in the controller. Create `app/controllers/settings/users_controller.rb` and add the following:
```

Comment:
Not a biggie, but I found it a bit confusing to mix the other settings (which seems mostly personal) with admin stuff. I'd suggest an admin namespace, but that would generate a lot of rework.

Reply:
I'm thinking a `Store` namespace here for Products, Users, and future features that admins can access.